### ISSUE TITLE
Include all MeshInstances in scene when exporting MeshLibrary (and not just the direct children of the scene root)

### ISSUE THREAD

**Issue by ScottyWired** [godotengine/godot-proposals#3406]
### Describe the project you are working on

A grid-based 3D turn-based-tactics game with large amounts of modular terrain. 

### Describe the problem or limitation you are having in your project

"Scene -> Convert To -> Meshlibrary..." only includes MeshInstance direct children of the scene root when creating the MeshLibrary.
When dragging a 3D model into the scene (in my case, a .glb file), it creates a scene instance consisting of a node (Spatial by default) with a MeshInstance child.

Each of these scene instances has to be "Make Local" and then the MeshInstance child reparented to the root. One by one. The MeshInstance child also needs renaming, as my 3D software (AssetForge) does not support naming meshes.
This is an extremely slow process for a game with hundreds of rapidly made terrain assets.
![image](https://user-images.githubusercontent.com/7548288/136655356-c883caec-8a39-4c79-a497-6413fc0bf36d.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow MeshLibrary export to include all MeshInstances in scene (including children inside other scenes) and not just MeshInstances which are direct children of the Scene Root.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![Godot_Engine_-_Tac3d_()_TNxxU copy](https://user-images.githubusercontent.com/7548288/136655011-0aa268f1-2a47-4ae7-9857-233c0acb73f2.png)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Import script that takes the mesh from the 3D file and gives the mesh the same name as the 3D file, then optionally deletes the 3D file. 

### Is there a reason why this should be core and not an add-on in the asset library?

![image](https://user-images.githubusercontent.com/7548288/136655125-5d65b164-3908-40a5-b7c4-505c2dc5d8c9.png)

**Comment by Calinou** [comment]
I don't think this should be made optional. Is there any reason you want non-direct children to be excluded from the conversion to MeshLibrary?

**Comment by ScottyWired** [comment]
"I don't think this should be made optional. Is there any reason you want non-direct children to be excluded from the conversion to MeshLibrary?"

You misunderstand, I want non-direct children to be **included**. Currently, only direct children are included, and I want non-direct children to be part of MeshLibraries.

"However, this requires you to have access to direct Mesh resources"

Yes that's the problem.

**Comment by Calinou** [comment]
> You misunderstand, I want non-direct children to be _included_. Currently, only direct children are included, and I want non-direct children to be part of MeshLibraries.

I was saying the new option you're proposing may not need to be made optional. We can make it the only behavior in 4.0 and later if there is no valid use case for the other behavior (including only direct children).

**Comment by oparisy** [comment]
I just tried something similar in Godot 4.0.2 and this ticket is spot on in its description of the desirable workflow and current limitations.

My idea was the following: copy a bunch of `glb` files into a folder in my project assets, let those auto-import, drag&drop those in a scene, export that scene as a `MeshLibrary`.

Simple enough, excepted that indeed imported `glb` are scenes, and drag&dropping leads to a scene not understood by the exporter to MeshLibrary (see [proposal 3821](https://github.com/godotengine/godot-proposals/issues/3821) for more details).

I see no way around this at this time except scripting a conversion to the proper scene structure, as described above. Manual reorganization of the scene is unpractical (101 assets, and for example "`Make Local`" does not work on multiple nodes selection).
